### ISSUE TITLE
I want static link crypto.a but has " Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking"

### ISSUE THREAD

**Issue by wangwenqiangGitHub** [openssl/openssl#19035]
```
openssl-OpenSSL_1_1_1h/build/lib/libcrypto.a(dso_dlfcn.o): in function `dlfcn_globallookup':
dso_dlfcn.c:(.text+0x17): warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libdl.a(dlopen.o): in function `dlopen':
(.text+0x9): undefined reference to `__dlopen'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libdl.a(dlclose.o): in function `dlclose':
(.text+0x5): undefined reference to `__dlclose'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libdl.a(dlsym.o): in function `dlsym':
(.text+0x9): undefined reference to `__dlsym'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libdl.a(dlerror.o): in function `dlerror':
(.text+0x5): undefined reference to `__dlerror'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libdl.a(dladdr.o): in function `dladdr':
(.text+0x5): undefined reference to `__dladdr'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/omUpgrade.dir/build.make:209: omUpgrade] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/omUpgrade.dir/all] Error 2
```

**Comment by paulidale** [comment]
Try configuring OpenSSL with the `no-shared` option.

**Comment by wangwenqiangGitHub** [comment]
> Try configuring OpenSSL with the `no-shared` option.

had add` no-shared ` but still this question

**Comment by wangwenqiangGitHub** [comment]
this is compile order:
`./config no-shared --prefix=/mnt/d/code/projectX86/openssl-OpenSSL_1_1_1h/build --openssldir=--prefix=/mnt/d/code/projectX86/openssl-OpenSSL_1_1_1h/build "-static -static-libgcc" `

**Comment by levitte** [comment]
First things first:

Why the quotes around `-static -static-libgcc`?  `Configure` knows how to detect `-static` if it's an argument of its own, and (more or less) disables the right stuff then.

**Comment by xtkoba** [comment]
The problem here is that the symbols `__dlopen` etc. referenced from `/usr/lib/x86_64-linux-gnu/libdl.a` is not resolvable by the toolchain used for building your application. I would expect that these symbols are defined in `/usr/lib/x86_64-linux-gnu/libc.a` which should be linked in unless any weird linker option (`-nolibc` or something) is used.

I guess either that your toolchain installation is broken or that the command used for linking your application is wrong. I don't believe either is relevant to OpenSSL in any way.

**Comment by nhorman** [comment]
marking as inactive, will be closed at the end of 3.4 dev barring further input
